imports/plugins/core/catalog/server/methods/catalog.js

```diff
@@ -986,11 +931,6 @@ Meteor.methods({
     check(field, String);
     check(value, Match.OneOf(String, Object, Array, Boolean, Number));
 
-    // Must have createProduct permission for active shop
-    if (!Reaction.hasPermission("createProduct")) {
-      throw new ReactionError("access-denied", "Access Denied");
-    }
-
     // Check first if Product exists and then if user has the right to alter it
```

Comment:
Good question. It's definitely best practice to avoid giving clues to existence for something like querying if a user with a certain email exists. In this case, though, the product ID is public information so we're not really revealing anything. The reason we look up the product first, is because we need `product.shopId` in order to do the permission check for the correct shop. We can't even do a permission check if we don't have a product. We could still choose to throw a less informative error, but as I said in this case we're not really revealing anything.

Reply:
👍 Makes sense, thanks for the clarification.